src/core/store/modules/vaults/vaults.actions.ts

```diff
@@ -192,12 +212,25 @@ const approveDeposit = createAsyncThunk<void, { vaultAddress: string; tokenAddre
 
 const approveZapOut = createAsyncThunk<void, { vaultAddress: string }, ThunkAPI>(
   'vaults/approveZapOut',
-  async ({ vaultAddress }, { dispatch }) => {
-    const ZAP_OUT_CONTRACT_ADDRESS = getConfig().CONTRACT_ADDRESSES.zapOut;
-    const result = await dispatch(
-      TokensActions.approve({ tokenAddress: vaultAddress, spenderAddress: ZAP_OUT_CONTRACT_ADDRESS })
-    );
-    unwrapResult(result);
+  async ({ vaultAddress }, { getState, extra }) => {
+    const { vaults, wallet, network } = getState();
+    const vaultData = vaults.vaultsMap[vaultAddress];
+    const { vaultService, transactionService } = extra.services;
+    const amount = extra.config.MAX_UINT256;
+
+    const accountAddress = wallet.selectedAddress;
+    if (!accountAddress) throw new Error('WALLET NOT CONNECTED');
+
+    const tx = await vaultService.approveZapOut({
+      network: network.current,
+      accountAddress,
+      amount,
+      vault: vaultData,
+    });
+
+    if (typeof tx !== 'boolean') {
+      await transactionService.handleTransaction({ tx: tx as TransactionResponse, network: network.current });
+    }
```

Comment:
same comment [here](https://github.com/yearn/yearn-finance-v3/pull/514/files#r838029524)  for this changes